docs/usage/configuration/sources/kubernetes.md

```diff
@@ -0,0 +1,123 @@
+# 
+
+---
+description: Ingests data through kubernetes node's and outputs `log` events.
+---
+
+<!--
+     THIS FILE IS AUTOGENERATED!
+
+     To make changes please edit the template located at:
+
+     scripts/generate/templates/docs/usage/configuration/sources/kubernetes.md.erb
+-->
+
+# kubernetes source
+
+![][assets.kubernetes_source]
+
+{% hint style="warning" %}
+The `kubernetes` source is in beta. Please see the current
+[enhancements][urls.kubernetes_source_enhancements] and
+[bugs][urls.kubernetes_source_bugs] for known issues.
+We kindly ask that you [add any missing issues][urls.new_kubernetes_source_issue]
+as it will help shape the roadmap of this component.
+{% endhint %}
+
+The `kubernetes` source ingests data through kubernetes node's and outputs [`log`][docs.data-model.log] events.
+
+## Example
+
+{% code-tabs %}
+{% code-tabs-item title="log" %}
+```javascript
+{
+    "host": "vector-agent-rmqbn",
+    "message": "<188>1 2019-10-16T17:50:44.216Z forwardproductize.info quisquam 1693 ID830 - Use the digital ADP capacitor, then you can calculate the solid state capacitor!\n",
+    "stream": "stdout",
+    "pod_uid": "default_flog-f8dd5f7b-tvgfn_52cdc270-c3e6-4769-b0a9-275481502618",
+    "container_name": "flog",
+    "timestamp": "2019-10-16T17:51:10.244625907Z"
+}
+```
+
+{% code-tabs %}
+{% code-tabs-item title="vector.toml" %}
+```coffeescript
+[sources.my_source_id]
+  type = "kubernetes" # must be: "kubernetes"
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+## Options
+
+
+
+## How It Works
+
+### Delivery Guarantee
+
+Due to the nature of this component, it offers a
+[**best effort** delivery guarantee][docs.guarantees#best-effort-delivery].
+
+### Deployment
+
+The `kubernetes` source is designed to be deployed to a Kubernetes cluster as a
+[`DaemonSet`][urls.kubernetes_daemonset]. You can find an [example config] in the
+`vector` repository. At a high level the `kubernetes` source will run an agent on
+each node and will internally use the file source to collect logs from `/var/log/pods`
+and a few other locations that Kubernetes places logs into. Via the [`DaemonSet`][urls.kubernetes_daemonset]
+Kubernetes will ensure that there is always a copy of the agent running on each node.
+
+### Environment Variables
+
+Environment variables are supported through all of Vector's configuration.
+Simply add `${MY_ENV_VAR}` in your Vector configuration file and the variable
+will be replaced before being evaluated.
+
+You can learn more in the [Environment Variables][docs.configuration#environment-variables]
+section.
+
+### Metadata
+
+Each event will contain a `message` field that contains the direct output from the containers stdout/stderr. There
```

Comment:
This isn't accurate, or the "Input/Output" section is not accurate because it does not contain the fields mentioned here. Which one is correct?